operator/pkg/helmreconciler/prune.go

```diff
@@ -78,6 +101,104 @@ func (h *HelmReconciler) Prune(manifests name.ManifestMap, all bool) error {
 	})
 }
 
+// PruneControlPlaneByRevisionWrapper is wrapper to remove specific control plane revision and update iop status.
+// It is used for the operator case.
+func (h *HelmReconciler) PruneControlPlaneByRevisionWrapper(ns, revision string) (*v1alpha1.InstallStatus, error) {
+	if ns == "" {
+		ns = istioDefaultNamespace
+	}
+	errStatus := &v1alpha1.InstallStatus{Status: v1alpha1.InstallStatus_ERROR}
+	pids, err := proxy.GetIDsFromProxyInfo("", "", revision, ns)
+	if err != nil {
+		return errStatus,
+			fmt.Errorf("failed to check proxy infos: %v", err)
+	}
+	// TODO(richardwxn): better handling of the warning status
+	if len(pids) != 0 {
+		return errStatus,
+			fmt.Errorf("there are proxies still pointing to the pruned control plane: %s",
+				strings.Join(pids, " "))
+	}
+	uslist, _, err := h.GetPrunedResourcesList(revision)
+	if err != nil {
+		return errStatus, err
+	}
+	if err := h.PruneControlPlaneByRevision(revision, uslist); err != nil {
+		return errStatus, err
+	}
+	return &v1alpha1.InstallStatus{Status: v1alpha1.InstallStatus_HEALTHY}, nil
+}
+
+// PruneControlPlaneByRevision removed resources of specific control plane revision.
```

Comment:
Added comment. I need to separate out part of the original function to GetPrunedResourcesList because that is needed to issue a to be pruned list at the beginning